### ISSUE TITLE
 unpack requires a bytes object of length 32 error

### ISSUE THREAD

**Issue by achapkowski** [GeospatialPython/pyshp#91]
Hello,

When I do the following:

```
shp = r"D:\data\dummytest.shp"
reader = shapefile.Reader(shp)
print(reader.fields)

```

The following error occurs:

```struct.error: unpack requires a bytes object of length 32```

This error occurs in ```__dbfHeader(self)```.

In addition, it appears that the package is also returning the incorrect number of fields.  There are 8 non spatial columns and it says there are 7.  I think it might be due to the fact that the shapefile has FID before the SPATIAL column, but it's only a guess.  The shapefile is empty (no records) and it was created using ArcGIS Desktop.

I'm using v1.2.10 with python 3.4.  Attached is the offending shapefile.

[dummytest.zip](https://github.com/GeospatialPython/pyshp/files/952423/dummytest.zip)

**Update**
It appears that if there are no records in the shapefile, the error will occur, but if you add a record, it will not throw this error.  Hope this helps!

**Comment by GeospatialPython** [comment]
Yes - the library doesn't do well with empty shapefiles.  Null shapefile types should work, but that's different than empty.  I can't think of a good use case where you'd want an empty shapefile lying around but let me know if there is one.  

Opening the shapefile in QGIS, it appears there are only 7 columns: "Id", "shortintfi", "longintfie", "floatfield", "doublefiel", "textfield", "datefield".  The "FID" you see in ArcGIS is a virtual construct created on the fly and not an actual column.  You can easily demonstrate this by deleting a feature.  As long as you  have the shapefile open in ArcMap, the FID column number will skip the missing feature.  But if you save the shapefile, remove it, and re-add it, ArcMap will renumber the FID column.

**Comment by karimbahgat** [comment]
A series of recent PRs and commits have fixed the problems associated with empty shapefiles (both read and write). I just tested the attached shapefile using the latest GitHub version and it worked without any problems. So until the next pip version just use the version from GitHub.

**Comment by achapkowski** [comment]
@karimbahgat thank you for the response.  Any idea when the next release will be pushed?

@GeospatialPython The simplest workflow is when an analysis workflow results in an empty shapefile, and that process is then used in a second process.  It can happen quite a bit.

**Comment by karimbahgat** [comment]
Since theres been a few fixes lately i agree that it would be timely with a new bugfix version, 1.2.11. Unless there are any reasons to wait, I just first want to fix #76 and merge #92 and then I can push the new version.

**Comment by karimbahgat** [comment]
@achapkowski version 1.2.11 is now pip installable.
